src/avalanchemq/client/client.cr

```diff
@@ -297,7 +305,7 @@ module AvalancheMQ
             @log.debug { "Discarding #{frame.class.name}, waiting for Close(Ok)" }
             frame.body.skip(frame.body_size)
           else
-            yield ch
+            @log.debug { "Discarding #{frame.inspect}, waiting for Close(Ok)" }
```

Comment:
This means that we ignore all frames coming in for a channel that's already been closed, before we could still process them, which seems wrong